x-pack/plugins/licensing/server/licensing_plugin.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { BehaviorSubject, Observable, Subscription, timer } from 'rxjs';
+import { first, map } from 'rxjs/operators';
+import moment from 'moment';
+import { CoreSetup, CoreStart, Logger, Plugin, PluginInitializerContext } from 'src/core/server';
+import { LicensingConfigType, LicensingPluginSubject, ILicensingPluginSetup } from './types';
+import { LicensingConfig } from './licensing_config';
+import { LicensingPluginSetup } from './licensing_plugin_setup';
+
+export class LicensingPlugin implements Plugin<LicensingPluginSubject> {
```

Comment:
There's a lot going on in this class. I think it might be better to move the polling code into a sub-service that this class simply wires up and exposes. This would make adding features to this class and testing this class much easier